iio/iio_ad9361/iio_ad9361.c

```diff
@@ -0,0 +1,2586 @@
+ * @len:	Maximum length of value to be stored in buf.
+ * @channel:	Channel properties.
+ * Return: Length of chars written in buf, or negative value on failure.
+ */
+static ssize_t get_sampling_frequency(void *device, char *buf, size_t len,
+				      const struct iio_ch_info *channel)
+{
+	struct ad9361_rf_phy *ad9361_phy = (struct ad9361_rf_phy *)device;
+	uint32_t sampling_freq_hz;
+	ssize_t ret = ad9361_get_rx_sampling_freq (ad9361_phy, &sampling_freq_hz);
+
+	if (ret < 0)
+		return ret;
+
+	return (ssize_t) snprintf(buf, len, "%"PRIi32"", sampling_freq_hz);
+}
+
+/**
+ * get_rf_bandwidth_available().
+ * @device:	Physical instance of a iio_axi_adc device.
+ * @buf:	Where value is stored.
+ * @len:	Maximum length of value to be stored in buf.
+ * @channel:	Channel properties.
+ * Return: Length of chars written in buf, or negative value on failure.
+ */
+static ssize_t get_rf_bandwidth_available(void *device, char *buf, size_t len,
+		const struct iio_ch_info *channel)
+{
+	if(channel->ch_out) {
+		return sprintf(buf, "[200000 1 40000000]");
+	} else {
+		return sprintf(buf, "[200000 1 56000000]");
+	}
+}
+
+/**
+ * get_rf_bandwidth().
+ * @device:	Physical instance of a iio_axi_adc device.
+ * @buf:	Where value is stored.
+ * @len:	Maximum length of value to be stored in buf.
+ * @channel:	Channel properties.
+ * Return: Length of chars written in buf, or negative value on failure.
+ */
+static ssize_t get_rf_bandwidth(void *device, char *buf, size_t len,
+				const struct iio_ch_info *channel)
+{
+	struct ad9361_rf_phy *ad9361_phy = (struct ad9361_rf_phy *)device;
+	if(channel->ch_out) {
+		return sprintf(buf, "%"PRIu32"", ad9361_phy->current_tx_bw_Hz);
+	} else {
+		return sprintf(buf, "%"PRIu32"", ad9361_phy->current_rx_bw_Hz);
+	}
+}
+
+/**
+ * get_gain_control_mode().
+ * @device:	Physical instance of a iio_axi_adc device.
+ * @buf:	Where value is stored.
+ * @len:	Maximum length of value to be stored in buf.
+ * @channel:	Channel properties.
+ * Return: Length of chars written in buf, or negative value on failure.
+ */
+static ssize_t get_gain_control_mode(void *device, char *buf, size_t len,
+				     const struct iio_ch_info *channel)
+{
+	struct ad9361_rf_phy *ad9361_phy = (struct ad9361_rf_phy *)device;
+	return (ssize_t) sprintf(buf, "%s",
+				 ad9361_agc_modes[ad9361_phy->agc_mode[channel->ch_num]]);
+}
+
+/**
+ * get_rf_dc_offset_tracking_en().
+ * @device:	Physical instance of a iio_axi_adc device.
+ * @buf:	Where value is stored.
+ * @len:	Maximum length of value to be stored in buf.
+ * @channel:	Channel properties.
+ * Return: Length of chars written in buf, or negative value on failure.
+ */
+static ssize_t get_rf_dc_offset_tracking_en(void *device, char *buf, size_t len,
+		const struct iio_ch_info *channel)
+{
+	struct ad9361_rf_phy *ad9361_phy = (struct ad9361_rf_phy *)device;
+
+	if(!channel->ch_out) {
+		buf[1] = 0;
+		return (ssize_t) sprintf(buf, "%"PRIi16"", ad9361_phy->rfdc_track_en) + 1;
+	}
```

Comment:
you must set buf[1] with 0 before sprintf, otherwise the value will be overwritten.